server/app/core/plugin_manager.py

```
@@ -0,0 +1,209 @@
+        for p in self.plugins_dir.iterdir():
+            if p.is_dir() and (p / "plugin.py").exists():
+                names.append(p.name)
+        return names
+
+    def load(self, name: str) -> None:
+        if name in self.loaded:
+            return
+        module_name = f"plugins.{name}.plugin"
+        mod = importlib.import_module(module_name)
+        if not hasattr(mod, "get_plugin"):
+            raise RuntimeError(f"Plugin {name} missing get_plugin()")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)